internals/src/Tizen.Peripheral/Tizen.Peripheral/Gpio.cs

```diff
@@ -111,39 +111,33 @@ public class GpioPin : IDisposable
         /// <param name="mode">GPIO direction.</param>
         public GpioPin(int pinNumber, GpioPinDriveMode mode)
         {
-            var ret = NativeGpio.Open(pinNumber, out IntPtr handle);
+            var ret = NativeGpio.Open(pinNumber, out _handle);
             if (ret != ErrorCode.None)
                 throw ExceptionFactory.CreateException(ret);
 
-            _handle = handle;
             PinNumber = pinNumber;
-            try
+            switch (mode)
             {
-                switch (mode)
-                {
-                    case GpioPinDriveMode.Input:
-                        ret = NativeGpio.SetEdgeMode(handle, NativeGpio.EdgeType.Both);
-                        if (ret != ErrorCode.None)
-                            throw ExceptionFactory.CreateException(ret);
-
-                        ret = NativeGpio.SetDirection(handle, NativeGpio.Direction.In);
-                        if (ret != ErrorCode.None)
-                            throw ExceptionFactory.CreateException(ret);
-                        SetIntteruptedCallback();
-                        break;
-                    case GpioPinDriveMode.OutputInitiallyLow:
-                        ret = NativeGpio.SetDirection(handle, NativeGpio.Direction.OutLow);
-                        if (ret != ErrorCode.None)
-                            throw ExceptionFactory.CreateException(ret);
-                        break;
-                    case GpioPinDriveMode.OutputInitiallyHigh:
-                        ret = NativeGpio.SetDirection(handle, NativeGpio.Direction.OutHigh);
-                        if (ret != ErrorCode.None)
-                            throw ExceptionFactory.CreateException(ret);
-                        break;
-                }
-            } finally {
-                Dispose();
+                case GpioPinDriveMode.Input:
+                    ret = NativeGpio.SetEdgeMode(_handle, NativeGpio.EdgeType.Both);
+                    if (ret != ErrorCode.None)
+                        throw ExceptionFactory.CreateException(ret);
+
+                    ret = NativeGpio.SetDirection(_handle, NativeGpio.Direction.In);
+                    if (ret != ErrorCode.None)
+                        throw ExceptionFactory.CreateException(ret);
+                    SetIntteruptedCallback();
+                    break;
+                case GpioPinDriveMode.OutputInitiallyLow:
+                    ret = NativeGpio.SetDirection(_handle, NativeGpio.Direction.OutLow);
+                    if (ret != ErrorCode.None)
+                        throw ExceptionFactory.CreateException(ret);
+                    break;
+                case GpioPinDriveMode.OutputInitiallyHigh:
+                    ret = NativeGpio.SetDirection(_handle, NativeGpio.Direction.OutHigh);
+                    if (ret != ErrorCode.None)
+                        throw ExceptionFactory.CreateException(ret);
+                    break;
```

Comment:
You need try and catch, because it is a constructor, if fail to constructor it does not create object so, developer can't release internal handle

``` c#
 try { 
 ....
 catch (Exception err) {
       Dispose();
       throw;
 }
```